File: native/zohodesk_portal_ticket/android/src/main/kotlin/com/zoho/desk/asap/ticket/zohodesk_portal_ticket/ZohodeskPortalTicketPlugin.kt
package com.zoho.desk.asap.ticket.zohodesk_portal_ticket

import android.app.Activity
import androidx.annotation.NonNull
import com.zoho.desk.asap.apikit.zohodesk_portal_apikit.ZDPBaseActivityAwarePlugin
import com.zoho.desk.asap.asap_tickets.ZDPortalTickets
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
import com.zoho.desk.asap.ticket.zohodesk_portal_ticket.data.ZDPTicketConfiguration
import com.google.gson.Gson

/** ZohodeskPortalTicketPlugin */

class ZohodeskPortalTicketPlugin: ZDPBaseActivityAwarePlugin("zohodesk_portal_ticket") {

  override fun showMethod(activity: Activity) {
    ZDPortalTickets.show(activity)
  }

  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: MethodChannel.Result) {
    if(call.method == "setConfiguration") {
      setConfiguration(call)
    }else{
      super.onMethodCall(call, result)
    }
  }

  private fun setConfiguration(@NonNull call: MethodCall) {
    val paramsMap = call.arguments as? HashMap<*, *>
    val configurationData = paramsMap?.get("configuration") as? String
    val ticketConfiguration: ZDPTicketConfiguration = Gson().fromJson(configurationData, ZDPTicketConfiguration::class.java)
    ZDPortalTickets.setConfiguration(
      com.zoho.desk.asap.asap_tickets.utils.ZDPTicketConfiguration.Builder()
        .isReplyAllowed(ticketConfiguration.isReplyAllowed)
        .isCommentAllowed(ticketConfiguration.isCommentAllowed)
        .isTicketUpdateAllowed(ticketConfiguration.isTicketUpdateAllowed)
        .isCommentEditAllowed(ticketConfiguration.isCommentEditAllowed)
        .isCommentDeleteAllowed(ticketConfiguration.isCommentDeleteAllowed)
        .isAddTicketAllowed(ticketConfiguration.isAddTicketAllowed)
        .isHappinessThreadAllowed(ticketConfiguration.isHappinessThreadAllowed)
        .isTicketPropertiesAllowed(ticketConfiguration.isTicketPropertiesAllowed)
        .isTicketChannelAllowed(ticketConfiguration.isTicketChannelAllowed)
        .isTicketDetailSearchAllowed(ticketConfiguration.isTicketDetailSearchAllowed)
        .build()
    )
  }

}

File: native/zohodesk_portal_ticket/ios/Classes/ZohodeskPortalTicketPlugin.swift
import Flutter
import UIKit
import ZohoDeskPortalTicket
import zohodesk_portal_apikit

public class ZohodeskPortalTicketPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "zohodesk_portal_ticket", binaryMessenger: registrar.messenger())
    let instance = ZohodeskPortalTicketPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
      let arguments: [String: Any]? = call.arguments as? [String: Any]
      switch ZDPTicketAPIs(rawValue: call.method) {
      case .show:
        ZDPortalTicket.show()
      case .setConfiguration:
        handleTicketConfigurations(arguments: arguments)
      default:
          break
      }
  }

  private func handleTicketConfigurations(arguments: [String: Any]?) {
      guard let configurationJSON = arguments?["configuration"] as? String else {
         return
      }
      if let ticketConfigurations: ZDFTicketConfiguration = configurationJSON.parser() {
          ZDPTicketConfiguration.configure.isReplyAllowed = ticketConfigurations.isReplyAllowed
          ZDPTicketConfiguration.configure.isCommentAllowed = ticketConfigurations.isCommentAllowed
          ZDPTicketConfiguration.configure.isTicketUpdateAllowed = ticketConfigurations.isTicketUpdateAllowed
          ZDPTicketConfiguration.configure.isCommentEditAllowed = ticketConfigurations.isCommentEditAllowed
          ZDPTicketConfiguration.configure.isCommentDeleteAllowed = ticketConfigurations.isCommentDeleteAllowed
          ZDPTicketConfiguration.configure.isAddTicketAllowed = ticketConfigurations.isAddTicketAllowed
          ZDPTicketConfiguration.configure.isCustomerHappinessThreadAllowed = ticketConfigurations.isHappinessThreadAllowed
          ZDPTicketConfiguration.configure.isTicketChannelAllowed = ticketConfigurations.isTicketChannelAllowed
          ZDPTicketConfiguration.configure.isTicketPropertiesAllowed = ticketConfigurations.isTicketPropertiesAllowed
          ZDPTicketConfiguration.configure.disableTicketDetailSearch = !ticketConfigurations.isTicketDetailSearchAllowed
      }
  } 
    
    private enum ZDPTicketAPIs: String {
        case show
        case setConfiguration
    }
}

private class ZDFTicketConfiguration: Decodable {
     var isReplyAllowed: Bool = true;
     var isCommentAllowed: Bool = true;
     var isTicketUpdateAllowed: Bool = true;
     var isCommentEditAllowed: Bool = true;
     var isCommentDeleteAllowed: Bool = true;
     var isAddTicketAllowed: Bool = true;
     var isHappinessThreadAllowed: Bool = true;
     var isTicketChannelAllowed = true
     var isTicketPropertiesAllowed = true
     var isTicketDetailSearchAllowed: Bool = true
}
